service/dynamodb/expression/condition.go

```diff
@@ -28,21 +54,76 @@ func (cm ConditionMode) String() string {
 		return "UnsetCond"
 	case EqualCond:
 		return "EqualCond"
+	case NotEqualCond:
+		return "NotEqualCond"
+	case LessCond:
+		return "LessCond"
+	case LessEqualCond:
+		return "LessEqualCond"
+	case GreaterCond:
+		return "GreaterCond"
+	case GreaterEqualCond:
+		return "GreaterEqualCond"
 	case AndCond:
 		return "AndCond"
+	case OrCond:
+		return "OrCond"
+	case NotCond:
+		return "NotCond"
 	case BetweenCond:
 		return "BetweenCond"
+	case InCond:
+		return "InCond"
+	case AttrExistsCond:
+		return "AttrExistsCond"
+	case AttrNotExistsCond:
+		return "AttrNotExistsCond"
+	case BeginsWithCond:
+		return "BeginsWithCond"
+	case ContainsCond:
+		return "ContainsCond"
 	default:
 		return "no matching ConditionMode"
 	}
 }
 
+// DynamoDBAttributeType will specify the type of an DynamoDB item attribute.
+// This enum will be used in the AttributeType() function in order to be
+// explicit about the DynamoDB type that is being checked and ensure compile
+// time checks
+// More Informatin at http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.OperatorsAndFunctions.html#Expressions.OperatorsAndFunctions.Functions
+type DynamoDBAttributeType string
+
+const (
+	// String will represent the DynamoDB String type
+	String DynamoDBAttributeType = "S"
+	// StringSet will represent the DynamoDB String Set type
+	StringSet DynamoDBAttributeType = "SS"
```

Comment:
The extra `DynamoDBAttributeType` are not needed on each line. its assumed by the compiler that all types are the same as the parent per const block.

e.g.

```go
const (
    String DynamoDBAttributeType = "S"
    StringSet  = "SS"
```